services/src/main/java/org/keycloak/services/resources/admin/IdentityProviderResource.java

```diff
@@ -133,10 +125,8 @@ public Response delete() {
         session.users().preRemove(realm, identityProviderModel);
         this.realm.removeIdentityProviderByAlias(alias);
 
-        Set<IdentityProviderMapperModel> mappers = this.realm.getIdentityProviderMappersByAlias(alias);
-        for (IdentityProviderMapperModel mapper : mappers) {
-            this.realm.removeIdentityProviderMapper(mapper);
-        }
+        realm.getIdentityProviderMappersByAliasStream(alias)
+                .collect(Collectors.toList()).forEach(realm::removeIdentityProviderMapper);
```

Comment:
Is it necessary to call collect here?

Reply:
Yes, otherwise it throws ConcurrentModificationException.

Reply:
Ah ok, thanks!